ocaml/xapi/xapi_pool.ml

```diff
@@ -1077,6 +1077,75 @@ let designate_new_master ~__context ~host =
     Xapi_pool_transition.attempt_two_phase_commit_of_new_master ~__context true peers my_address
   end
 
+let management_reconfigure ~__context ~network =
+  (* Disallow if HA is enabled *)
+  let pool = Helpers.get_pool ~__context in
+  if Db.Pool.get_ha_enabled ~__context ~self:pool then
+    raise (Api_errors.Server_error(Api_errors.ha_is_enabled, []));
+
+  (* Disallow if network provided is Host internal management network *)
+  let network_other_config = Db.Network.get_other_config ~__context ~self:network in
+  if (List.mem_assoc Xapi_globs.is_host_internal_management_network network_other_config) &&
+    (List.assoc Xapi_globs.is_host_internal_management_network network_other_config = "true") then
+    raise (Api_errors.Server_error(Api_errors.pool_management_cannot_be_set_on_system_network, [Ref.string_of network]));
+
+  (* Create a hash table for hosts with pifs for the network *)
+  let pifs_on_network = Db.Network.get_PIFs ~__context ~self:network in
+  let hosts_with_pifs = Hashtbl.create 16 in
+  List.iter (fun self ->
+    let host = Db.PIF.get_host ~__context ~self in
+    Hashtbl.add hosts_with_pifs host self;
+  ) pifs_on_network;
+
+  (* All Hosts must have associated PIF on the network *)
+  let all_hosts = Db.Host.get_all ~__context in
+  List.iter (fun host ->
+    if not(Hashtbl.mem hosts_with_pifs host) then
+      raise (Api_errors.Server_error(Api_errors.pif_not_present, [Ref.string_of host; Ref.string_of network]));
+  ) all_hosts;
+
+  (* PIFs must not be bond slaves *)
+  List.iter (fun self ->
+    if Db.PIF.get_bond_slave_of ~__context ~self <> Ref.null then
+      raise (Api_errors.Server_error (Api_errors.cannot_plug_bond_slave, [Ref.string_of self]))
+  ) pifs_on_network;
+
+  (* PIFs must have same primary_address_type *)
+  (* PIFs must not have ip_configuration_mode as None *)
+  let address_type = Db.PIF.get_primary_address_type ~__context ~self:(List.hd pifs_on_network) in
+  if address_type = `IPv4 then
+    List.iter (fun self ->
+      if Db.PIF.get_primary_address_type ~__context ~self <> `IPv4 then
+        raise (Api_errors.Server_error(Api_errors.pif_incompatible_primary_address_type, [Ref.string_of self]));
+      if Db.PIF.get_ip_configuration_mode ~__context ~self = `None then
+        raise (Api_errors.Server_error(Api_errors.pif_has_no_network_configuration, [Ref.string_of self]));
+    ) pifs_on_network
+  else
+    List.iter (fun self ->
+      if Db.PIF.get_primary_address_type ~__context ~self <> `IPv6 then
+        raise (Api_errors.Server_error(Api_errors.pif_incompatible_primary_address_type, [Ref.string_of self]));
+      if Db.PIF.get_ipv6_configuration_mode ~__context ~self = `None then
+        raise (Api_errors.Server_error(Api_errors.pif_has_no_v6_network_configuration, [Ref.string_of self]));
+    ) pifs_on_network;
+
+  (* Perform Host.management_reconfigure on slaves first and last on master *)
```

Comment:
It would be cleaner to use the existing function
`Xapi_pool_helpers.call_fn_on_slaves_then_master`